src/utils/helpers.coffee

```diff
@@ -64,6 +65,15 @@ exports.excludeFile = (fileName, options) ->
             # Only instrument files that are inside the project.
             excluded = true
 
+        # For each exclude value try to use it as a pattern to exclude files
+        exclude.map (pattern) ->
+            glob.sync pattern,
```

Comment:
Now I get it. And thinking about it I realized that I was thinking about the problem all wrong: If we're just _excluding_ files there is no need to hit the filesystem at all to make the comparison, which should significantly speed up the globbing process and allow for something like what you just suggested.
I'll be working on a PR on one of the next days :)